uswds/uswds
# USWDS: Improve prettier and lint scripts

## Summary
Updated `package.json` by adding scripts to run Prettier on JavaScript files and updating the `lint` script to point to the correct directory. 

This PR also remediates the issues found by prettier and ESLint.

## Breaking change
This is not a breaking change.

## Related issue
Closes #4850 

## Preview link
Preview link: N/A

## Problem statement
USWDS needs a way to format and lint its Javascript files. 

### ESLint
`npm run lint` currently points to the outdated `src` directory. Instead, it should point to the `packages` directory added in 3.0. 

Once the directory was updated, only one error was reported by ESLint. This was fixed in https://github.com/uswds/uswds/pull/5109/commits/fff9977eaa79662dba17bd75863623b757282e28:
```
❯ npm run lint                            

> @uswds/uswds@3.3.0 lint
> eslint packages/**/**/*.js

/Users/aleadem/Dev/github/uswds-workspace/uswds/packages/usa-accordion/src/index.js
  71:16  error  'event' is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

## Solution
Updated `package.json` to:
1. Create a new `prettier:js` script for running Prettier on our Javascript files.
1. Create a new `prettier` script for running Prettier on HTML, JS, and SCSS files.
1. Updated the directory for the `lint` script to look in `packages` instead of outdated `src`.

## Testing and review
To test:
- Confirm that both `npm run prettier:js` and `npm run lint` run on the necessary Javascript files and that there are no errors reported. 
- Confirm that `npm run prettier` runs on the correct files.
- Confirm that running these scripts results in no errors, warnings, or changes.
